ui/helpers/utils/util.js

```diff
@@ -631,3 +633,40 @@ export const getNetworkNameFromProviderType = (providerName) => {
 export const isAbleToExportAccount = (keyringType = '') => {
   return !keyringType.includes('Hardware') && !keyringType.includes('Snap');
 };
+
+/**
+ * Checks if a tokenId in Hex or decimal format already exists in an object.
+ *
+ * @param {*} address - collection address.
+ * @param {*} tokId - tokenId to search for
+ * @param {*} obj - object to look into
+ * @returns {boolean} `false` if tokenId does not already exist.
+ */
+export const tokenIdExist = (address, tokId, obj) => {
+  // check if input tokenId is hexadecimal
+  // If it is convert to decimal and compare with existing tokens
+  // if it is decimal convert to hexadecimal and compare
+  const isHex = isStrictHexString(tokId);
+  let convertedTokenId;
+  if (isHex) {
+    // Convert to decimal
+    convertedTokenId = parseInt(tokId, 16);
+  } else {
+    // Convert to hex
+    const decimalNumber = parseInt(tokId, 10); // 10 is the base for decimal
+    convertedTokenId = `0x${decimalNumber.toString(16)}`;
+  }
+
+  if (obj[address]) {
+    const value = obj[address];
+    return lodash.some(value.nfts, (nft) => {
+      return (
+        nft.address === address &&
+        (nft.tokenId.toLowerCase() === tokId.toLowerCase() ||
+          nft.tokenId.toLowerCase() ===
+            convertedTokenId.toString().toLowerCase())
```

Comment:
Should this comparison have a chain id?  Theoretically there could be separate NFTs with the same address and token ID but different networks.  But I don't know whether this NFT state is *already* scoped to the network you're on.

Reply:
I think they are already scoped by the network you on; but i will  double check 🙏

Reply:
Resolving this as its scoped by network 🙏